RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitSoAToLegacy.cc

```diff
@@ -0,0 +1,96 @@
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHit.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitHostCollection.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/EmptyGroupDescription.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+
+class EcalRecHitSoAToLegacy : public edm::stream::EDProducer<> {
+public:
+  explicit EcalRecHitSoAToLegacy(edm::ParameterSet const &ps);
+  ~EcalRecHitSoAToLegacy() override = default;
+  static void fillDescriptions(edm::ConfigurationDescriptions &);
+
+private:
+  using InputProduct = EcalRecHitHostCollection;
+  void produce(edm::Event &, edm::EventSetup const &) override;
+
+private:
+  const bool isPhase2_;
+  const edm::EDGetTokenT<InputProduct> recHitsPortableEB_;
+  const edm::EDGetTokenT<InputProduct> recHitsPortableEE_;
+  const edm::EDPutTokenT<EBRecHitCollection> recHitsCPUEBToken_;
+  const edm::EDPutTokenT<EERecHitCollection> recHitsCPUEEToken_;
+};
+
+void EcalRecHitSoAToLegacy::fillDescriptions(edm::ConfigurationDescriptions &confDesc) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("recHitsPortableEB", edm::InputTag("ecalRecHitPortable", "EcalRecHitsEB"));
+  desc.add<std::string>("recHitsLabelCPUEB", "EcalRecHitsEB");
```

Comment:
This follows the nomenclature in `EcalUncalibRecHitSoAToLegacy.cc`. The CPU means that this is for the legacy collection which is consumed by modules running on CPU.
Could be removed if people think it is cleaner without it.

Reply:
Thanks. I was confused because the input SoA is already on the CPU memory, whereas this parameter is about the label of the output EB collection. To me already the module class name `EcalRecHitSoAToLegacy` tells that the output are the legacy collections (so the "legacy" would not really have to be repeated in the parameter name, although I would not be against either). The `recHits` part of the parameter name is somewhat redundant, because both the input and output are "RecHits", just in different format. I think the missing part for me is the "label of output product" part, so I'd suggest to make that aspect more clear.